### PR TITLE
Private events

### PR DESCRIPTION
**Pull request by elpotatoslayer**
### In this project, I built an application named Private Events and that behaves similarly to Eventbrite. The main goal is to put in practice the main concepts of Associations in rails.

I made sure that:

- all foreign keys have corresponding indexes in the Database, as it will make your database performance better.

- test our code using Rspec and Capybara

### CODE REVIEW

**Comment by elpotatoslayer** [comment]
I changed the project repository because the codebase was messed up and created a new repo and started the project a fresh. This is the [link](https://github.com/Ralph-1/Private__events/pull/2) to the old pull request

File: app/views/events/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <div class="col-md-6 offset-3">
+        <% if event.errors.any? %>
+          <div id="error_explanation">
+            <h2><%= pluralize(event.errors.count, "error") %> prohibited this event from being saved:</h2>
+            <ul>
+              <% event.errors.full_messages.each do |message| %>
+                <li><%= message %></li>
+              <% end %>
+            </ul>
+          </div>
```
Comment (acushlakoncept) [inline app/views/events/_form.html.erb:13]: There should be absolutely NO logic in views.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          <% if current_user %>
+            <%= link_to "Signed In as " + @user.name, user_path(@user),class: 'text-white ml-3'  %>
+            <%= link_to 'All events', events_path, class: 'btn btn-light my-2 my-sm-0 text-info ml-2' %>
+            <%= link_to 'Create event', new_event_path, class: 'btn btn-light my-2 my-sm-0 text-info ml-3' %>
+            <%= link_to "Sign out", sign_out_path, method: :delete, class: 'btn btn-light my-2 my-sm-0 text-info ml-2' %>
+          <% else %>
+            <%= link_to 'All events', events_path, class: 'btn btn-light my-2 my-sm-0 text-info ml-2' %>
+            <%= link_to "Sign in", sign_in_path, class: 'text-white ml-3' %>
+            <%= link_to "Sign up", sign_up_path, class: 'text-white ml-3' %>
+          <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb:35]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @Ralph-1, great job here, the application works just as expected and your UI looks great. 👏 
Well done on writing tests, good coverage 👍 
Let's make a few changes, and we will be done.

- [ ] Handle validation errors properly in your controller as entering a location below the minimum requirement throws an error. Also note 5 is too much for minimum, Example name of the city I live is called Uyo, meaning I can't enter that as a valid location. 

- [ ] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          <div class="card-header bg-primary font-weight-bold h3">
+            Event Details
+          </div>
+          <%= link_to event_path(event) do %>
+          <ul class="list-group list-group-flush">
+            <li class="list-group-item">- <strong>Date:</strong> <%= event.date %></li>
+            <li class="list-group-item">- <strong>Location:</strong> <%= event.location %></li>
+            <li class="list-group-item">- <strong>Description:</strong> <%= event.description %></li>
+          </ul>
+          <% end %>
```
Comment (peterrobert) [inline app/views/events/index.html.erb:21]: - [ ] There should absolutely be no logic in your views. Views are only for displaying information that became available to it from its corresponding controller. Kindly use this [resource](http://codefol.io/posts/Where-Do-I-Put-My-Code/) to learn more on this.

- Kindly use the rails helpers for your logic. Logic includes, Loops and If statements.

File: app/views/events/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  </div>
+  <%# Past Events %>
+  <h3 class="text-center text-muted">Past events</h3>
+  <hr/>
+  <div class="row">
+    <% @past_events.each do |event| %>
+      <div class="col-md-4">
+        <div class="card my-2">
+          <div class="card-header bg-info font-weight-bold h3">
+            Event Details
```
Comment (peterrobert) [inline app/views/events/index.html.erb:35]: - [ ] There should absolutely be no logic in your views. Views are only for displaying information that became available to it from its corresponding controller. Kindly use this [resource](http://codefol.io/posts/Where-Do-I-Put-My-Code/) to learn more on this.

- Kindly use the rails helpers for your logic. Logic includes, Loops and If statements.

File: app/views/events/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<section class="event my-4">
+<h2>Belongs to: <%= @event.creator.name %></h2>
+<h5 class="text-primary"><span class="text-secondary">Event:</span>  <%= @event.location %> - <%= @event.date %></h5>
+<section class="attendants">
+  <h3 class="h4">People attending/attended this event:</h3>
+  <ul class="attendants-list">
+  <% @event.attendees.each do |attendee| %>
+    <li><%= attendee.name %></li>
+  <% end %>
```
Comment (peterrobert) [inline app/views/events/show.html.erb:9]: - [ ] There should absolutely be no logic in your views. Views are only for displaying information that became available to it from its corresponding controller. Kindly use this [resource](http://codefol.io/posts/Where-Do-I-Put-My-Code/) to learn more on this.

- Kindly use the rails helpers for your logic. Logic includes, Loops and If statements.

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED ♻️ 🛑 
Hello, @Ralph-1  👋🏿 . Great work making the changes that were requested by the previous TSE. Also, great work with the application. It almost meet all the requirements. However, for your project to be approved, you need to do some additional changes kindly check below ⬇️ 

- [ ] Kindly remove all the logic from your views. The views should only be for displaying information. Any logic should be created in the helpers.

### if you have any question kindly reach me on slack @peterrobert


### HAPPY CODING 💻

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

### Hi, @Ralph-1  well done on this project, I am sure you're feeling more and more confident in tackling Ruby-on-Rails project.

I will go ahead and mark this as complete, feel free to merge when you are ready. All the best with the next project.


### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**